arbnode/transaction_streamer.go

```diff
@@ -1292,113 +1291,96 @@ func (s *TransactionStreamer) executeMessages(ctx context.Context, ignored struc
 	}
 
-	nextHeader, err := s.espressoClient.FetchHeaderByHeight(ctx, snapshot.Height)
+	batch := s.db.NewBatch()
+	err = s.setEspressoSubmittedTxns(batch, newSubmittedTxns)
 	if err != nil {
-		return fmt.Errorf("error fetching the snapshot header (height: %d): %w", snapshot.Height, err)
+		return fmt.Errorf("failed to set espresso submitted txns: %w", err)
 	}
 
-	proof, err := s.espressoClient.FetchBlockMerkleProof(ctx, snapshot.Height, height)
-	if err != nil {
-		return fmt.Errorf("error fetching the block merkle proof (height: %d, root height: %d): %w", height, snapshot.Height, err)
+	if err = batch.Write(); err != nil {
+		return fmt.Errorf("failed to write to db: %w", err)
 	}
 
-	blockMerkleTreeRoot := nextHeader.Header.GetBlockMerkleTreeRoot()
-
-	ok := espressocrypto.VerifyMerkleProof(proof.Proof, jsonHeader, *blockMerkleTreeRoot, snapshot.Root)
-	if !ok {
-		return fmt.Errorf("error validating merkle proof (height: %d, snapshot height: %d)", height, snapshot.Height)
-	}
+	return nil
+}
 
-	// Verify the namespace proof
-	resp, err := s.espressoClient.FetchTransactionsInBlock(ctx, height, s.chainConfig.ChainID.Uint64())
+func (s *TransactionStreamer) checkEspressoQueryNodesForTransasction(ctx context.Context, hash *tagged_base64.TaggedBase64) (espressoTypes.TransactionQueryData, error) {
+	tx, err := s.espressoClient.FetchTransactionByHash(ctx, hash)
 	if err != nil {
-		return fmt.Errorf("failed to fetch the transactions in block (height: %d): %w", height, err)
-	}
-
-	namespaceOk := espressocrypto.VerifyNamespace(
-		s.chainConfig.ChainID.Uint64(),
-		resp.Proof,
-		*header.Header.GetPayloadCommitment(),
-		*header.Header.GetNsTable(),
-		resp.Transactions,
-		resp.VidCommon,
-	)
-
-	if !namespaceOk {
-		return fmt.Errorf("error validating namespace proof (height: %d)", height)
+		return espressoTypes.TransactionQueryData{}, fmt.Errorf("failed to fetch transaction from espresso: %w", err)
 	}
 
-	submittedPayload := firstSubmitted.Payload
-	validated := arbutil.ValidateIfPayloadIsInBlock(submittedPayload, resp.Transactions)
-	if !validated {
-		return fmt.Errorf("transactions fetched from HotShot doesn't contain the submitted payload")
-	}
-
-	// Reset the last submit failure time if we successfully fetch the transaction and verify its inclusion/namespace proof
-	s.lastSubmitFailureAt = nil
-
-	// Validation completed. Update the database
-	s.espressoTxnsStateInsertionMutex.Lock()
-	defer s.espressoTxnsStateInsertionMutex.Unlock()
+	return tx, nil
+}
 
-	batch := s.db.NewBatch()
-	if err := s.setEspressoSubmittedTxns(batch, submittedTxns[1:]); err != nil {
-		return fmt.Errorf("failed to set the espresso submitted txns: %w", err)
+func (s *TransactionStreamer) resubmitTransactionIfPastDelay(ctx context.Context, submittedTx arbutil.SubmittedEspressoTx) (*arbutil.SubmittedEspressoTx, error) {
+	timeSinceSubmission := time.Since(submittedTx.SubmittedAt)
+	if timeSinceSubmission <= s.resubmitEspressoTxDeadline {
```

Comment:
I think this should be if timeSinceSubmission = s.resubmitEspressoTxDeadline {

It if equal to the deadline, we should resubmit